mantidproject/mantid
# Rename AsciiSaver Class

### Description of work

#### Summary of work
External references to SaveAscii were renamed in the Refl GUI when the ORSO nexus format was added. This PR just finishes that work to update the internal class, file, and test names. 

Fixes #37292

#### Further detail of work
- `FileSaver` was selected to be the new name for these classes. 

### To test:
Checking that the ORSO saving still performs as normal. 
1) Go to `Interfaces` -> `Reflectometry` -> `ISIS Reflectometry`.
1) Load the example batch from the ISIS Sample Data (INTER_NR_test)
1) Go to the Save tab. Set a Save Path for saving the file to.
1) In the File Format section, select `ORSO Nexus (*.orb)` from the drop-down.
1) In the Automatic Save section, tick the `Save automatically on completion` checkbox.
1) The `Include individual row outputs for groups` checkbox should be enabled. Select this checkbox.
1) The `Save multiple datasets to a single file` checkbox should be enabled and selected by default.
1) Return to the runs tab and process all.
1) If you navigate to the save path that you set in the Save tab, you should see `.orb` files.
1) Test that the checkboxes on the Save tab are enabled and disabled correctly for the ORSO Nexus format (checkbox availability and behaviour should be the same as for the ORSO Ascii format - see manual testing documentation changes in this PR for further details, if required).
1) Swap to any of the other formats (which are ASCII), clear the ADS, and re-process. The chosen filetype should be produced. 


*This does not require release notes* because **there should be no user-facing changes to the behaviour.**


---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
